include/maths/CBoostedTreeLeafNodeStatistics.h

```diff
@@ -559,37 +552,35 @@ class MATHS_EXPORT CBoostedTreeLeafNodeStatistics {
         }
 
         //! Get the reduction of the per thread aggregate derivatives.
-        CSplitsDerivatives& reducedDerivatives(std::size_t index) {
+        CSplitsDerivatives& reducedDerivatives() {
```

Comment:
At one point I thought I would need to have separate derivatives on old and new training data. So I introduced this in an earlier change on the feature branch. In fact, it turned out that this can all be transparent from this code and handled by the loss functions. So this is just reverting an unnecessary complication (the supplied index was always 0).